### DISCUSSION TITLE
220516 아리 인터뷰 피드백

### DISCUSSION THREAD

**Discussion by YJGwon** [woowacourse-study/2022-lv2-effective-java-interview#8] [General]
### 인터뷰어 -> 인터뷰이

> 점수를 매겨주셔도 좋고, 체크 형식으로 작성해주신 뒤 짧은 코멘트를 남겨주셔도 좋습니다. 해당 항목에 대해 자유롭게 의견을 남겨주세요.

- 말하기 태도, 어조가 적절했나요?
- 전달하고자 하는 바를 명료하게 전달했나요?
- 본인만의 주관이 잘 드러났나요?
- 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- 모르는 부분에 대해 잘 대처했나요?

### 인터뷰이 -> 인터뷰어

> 인터뷰어로써의 역량을 키우려는 것이 아닌 저희 스터디의 원활한 진행을 위한 피드백입니다. 각 크루가 아닌 전반적인 진행 상황에 대해 의견을 남겨주세요!

- 사전에 공유한 내용이 잘 전달되었나요?
- 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
- 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
- 새로운 관점을 배울 수 있었나요?

**Comment by ulimy** [comment]
단순히 질문만 하고 끝나는 것이 아니라 자세한 설명까지 덧붙여주셔서 개인적인 성장에 많은 도움이 되었습니다! 덕분에 알게된 지식들 앞으로 요긴하게 잘 써먹겠습니다..ㅎㅎㅎ

**Comment by yeon-06** [comment]
많이 떨렸는지 발표는 말이 좀 빨랐는데 인터뷰 할 때는 침착하게 잘 해주신 것 같아요^ㅡ^)b
자신이 경험한 점과 경험해보지 못한 점을 솔직하게 공유해주신 점이 인상 깊었습니다
생각 말씀해주실 때 오 저렇게도 표현할 수 있구나 싶은 말들도 있어서 잘 들었어요

**Comment by tco0427** [comment]
- 말하기 태도, 어조가 적절했나요?
  -> 말하기 속도가 조금 빠른 부분을 제외하면 적절했다고 생각한다.
- 전달하고자 하는 바를 명료하게 전달했나요?
  -> 네
- 본인만의 주관이 잘 드러났나요?
  -> 본인이 알고 있는 내용을 바탕으로 충분히 잘 이야기해주셨습니다.
- 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
  -> 책의 내용과는 약간의 차이가 있지 않았나 하는 생각이 듭니다.
- 모르는 부분에 대해 잘 대처했나요?
  -> 모르는 부분은 확실하게 모른다고 이야기해주었고, 아는 내용은 아는대로 잘 이야기해주셨습니다.

**Comment by YJGwon** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
- [x] 전달하고자 하는 바를 명료하게 전달했나요?
- [ ] 본인만의 주관이 잘 드러났나요?
- [ ] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- [x] 모르는 부분에 대해 잘 대처했나요?

쫄지 않고 당차게 명랑하게 답변을 이어나가는 모습이 멋있었습니다! 잘 모르겠는 부분에 대해서도 텐션 잃지 않고 또박또박 생각을 잘 이야기해주셨어요. 다만 책에서 소개한 개념들에 대해 깊이있는 이해가 이루어졌다는 느낌은 가지기 어려웠습니다. 어떤 api여도 고정적인 용도라는 것은 없다고 생각해요. 아리만의 기준을 가지고 장단점을 가려봤으면 좋겠습니다!

**Comment by kyukong** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
- [x] 전달하고자 하는 바를 명료하게 전달했나요?
- [x] 본인만의 주관이 잘 드러났나요?
- [x] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- [x] 모르는 부분에 대해 잘 대처했나요?

말의 속도는 빠른 편이었으나 발음이 정확하여 듣기 어렵지 않았습니다. 오히려 빠르고 정확하게 말해주어 이해하기 쉬웠어요. 과일로 예시를 들어준 것도 좋았습니다. 모르는 것을 모른다고 솔직하게 말하고 다른 크루가 공부한 내용을 습득하는 것도 좋았습니다. 덕분에 저도 얻어갔네요.ㅎㅎㅎ 수고하셨어요!

**Comment by YJGwon** [comment]
### 질문

- getter 반환값 이외에 방어적 복사가 필요한 경우?
- 객체 내부의 참조값은 항상 노출되면 안될까?
- 3가지방법 중 아리가 선택한 방법?
- copyOf가 아닌 unmodifiableList를 선택하는게 좋을 경우는 없을까?
- 객체가 안전다면 깊은복사 X, 객체가 안전하다고 느끼는 기준?
- 방어적 복사가 getter의 반환값을 변경할 수 없도록 하기 위함이라면, getter의 반환값이 변경되지 않아야 할 이유는?
- “객체가 변하지 않도록 잘 설계”되지 못했을 때에는 항상 깊은 복사를 구현할 것인가?
- 깊은 복사 또는 방어적 복사를 사용하지 않고 객체의 값을 지키기 위해 노력해본 경험?
- 깊은 복사를 안하는게 좋은 이유?
- “객체가 변하지 않도록 잘 설계”한다는 것은 어떤 설계일까
- 외부에서 들어올 때, 내부에서 내보낼 때 어떻게 처리해야 할지 기준을 가지면 좋겠다

**Comment by Ohzzi** [comment]
말하기 태도, 어조가 적절했나요?
  - 말이 굉장히 빠릅니다. 속도를 꽤나 많이 줄이는 것이 도움이 될 것 같아요!

전달하고자 하는 바를 명료하게 전달했나요?
  - 말이 빠름에도 불구하고 딕션이 좋아서 말하기 속도에 비해 이해가 잘 되었습니다! 속도만 조절하면 훨씬 듣기 좋을 것 같아요!

본인만의 주관이 잘 드러났나요?
  - 본인의 생각과 기준이 확실히 있었습니다!

책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
  - 책보다 본인의 경험을 기준으로 이야기 잘 해주신 것 같아요!

모르는 부분에 대해 잘 대처했나요?
  - 모르는 부분에 대해 수용하는 자세를 잘 보여줬습니다!
